src/api/index.test.js

```diff
@@ -581,34 +696,228 @@ it('getNamespaces', () => {
     items: 'namespaces'
   };
   fetchMock.get(/namespaces/, data);
-  return getNamespaces().then(response => {
+  return index.getNamespaces().then(response => {
     expect(response).toEqual(data.items);
     fetchMock.restore();
   });
 });
 
+it('getServiceAccount', () => {
+  const data = { items: 'serviceaccounts' };
+  fetchMock.get(/serviceaccounts/, data);
+  return index.getServiceAccount('default', 'default').then(response => {
+    expect(response).toEqual(data);
+    fetchMock.restore();
+  });
+});
+
 it('getServiceAccounts', () => {
   const data = { items: 'serviceaccounts' };
   fetchMock.get(/serviceaccounts/, data);
-  return getServiceAccounts().then(response => {
+  return index.getServiceAccounts().then(response => {
     expect(response).toEqual(data.items);
     fetchMock.restore();
   });
 });
 
+it('getSecretServiceAccountList No Secrets Matched', () => {
+  const dataFail = [];
+
+  return index
+    .getSecretServiceAccountList(dataFail, 'secret-name')
+    .then(response => {
+      expect(response).toEqual([]);
+    });
+});
+
+it('getSecretServiceAccountList No Secrets Matched', () => {
+  const data = serviceAccountFail;
+  return index
+    .getSecretServiceAccountList(data, 'secret-name')
+    .then(response => {
+      expect(response).toEqual([]);
+    });
+});
+
+it('getSecretServiceAccountList 1 Secret Matched', () => {
+  const data = [serviceAccountPass];
+
+  return index
+    .getSecretServiceAccountList(data, 'secret-name')
+    .then(response => {
+      expect(response).toEqual(data);
+    });
+});
+
+it('getSecretServiceAccountList 2 Secret Matched', () => {
+  const data = [serviceAccountPass];
+
+  return index
+    .getSecretServiceAccountList(data, 'secret-name')
+    .then(response => {
+      expect(response).toEqual(data);
+    });
+});
+
+it('getSecretServiceAccountList 2 Secret Matched', () => {
+  const data = [serviceAccountPass, serviceAccount1, serviceAccount2];
+
+  const dataExpected = [serviceAccountPass, serviceAccount1];
+
+  return index
+    .getSecretServiceAccountList(data, 'secret-name')
+    .then(response => {
+      expect(response).toEqual(dataExpected);
+    });
+});
+
+it('getCredential', () => {
+  const credentialId = 'foo';
+  const data = { fake: 'credential' };
+  fetchMock.get(`end:${credentialId}`, data);
+  return index.getCredential(credentialId).then(credential => {
+    expect(credential).toEqual(data);
+    fetchMock.restore();
+  });
+});
+
+it('getIndexOfSecret secret exists', () => {
+  const data = [
+    {
+      name: 'default-token-dcb4b'
+    },
+    {
+      name: 'secret-name'
+    }
+  ];
+  const response = index.getIndexOfSecret(data, 'secret-name');
+  expect(response).toEqual(1);
+});
+
+it('getIndexOfSecret secret does not exist', () => {
```

Comment:
Test names should be of the format *verb* causes *action*, these are just collections of words atm